### PR TITLE
[PWX-25354] Set default cloud storage spec on EKS

### PR DESCRIPTION
**Pull request by pure-jliao**
**What this PR does / why we need it**:
* Set default eks cloud storage specs in `SetPortworxDefaults`
* As storage node per zone is needed, move max storage node per zone logic from controller API `setStorageClusterDefaults` (before set eks default) to Portworx API `SetDefaultsOnStorageCluster` (after set eks default)
* Fix unit test cases for changes above
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:
closed https://github.com/libopenstorage/operator/pull/808 and point target to master instead of mpxe feature branch

### CODE REVIEW

**Review by Pure-AdamuKaapan — APPROVED** [review]
LGTM!

File: drivers/storage/portworx/portworx.go [hunk 1]
```diff
@@ -679,18 +854,123 @@ func SetPortworxDefaults(toUpdate *corev1.StorageCluster, k8sVersion *version.Ve
...
+	toUpdate.Spec.Nodes = updatedNodeSpecs
+}
+
+func setPortworxStorageSpecDefaults(toUpdate *corev1.StorageCluster) {
+	if toUpdate.Spec.Kvdb == nil {
+		toUpdate.Spec.Kvdb = &corev1.KvdbSpec{}
+	}
+	if len(toUpdate.Spec.Kvdb.Endpoints) == 0 {
+		toUpdate.Spec.Kvdb.Internal = true
+	}
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx.go:973]: Technically this is not part of storage spec. If you want you can extract it out in `setKvdbDefaults()`.

File: drivers/storage/portworx/portworx.go [hunk 2]
```diff
@@ -712,14 +998,27 @@ func SetPortworxDefaults(toUpdate *corev1.StorageCluster, k8sVersion *version.Ve
...
-	if pxutil.IsTelemetryEnabled(toUpdate.Spec) && t.pxVersion.LessThan(pxutil.MinimumPxVersionCCM) {
-		toUpdate.Spec.Monitoring.Telemetry.Enabled = false // telemetry not supported for < 2.8
-		toUpdate.Spec.Monitoring.Telemetry.Image = ""
+func setPortworxCloudStorageSpecDefaults(toUpdate *corev1.StorageCluster) {
+	if preflight.IsEKS() {
+		if toUpdate.Spec.CloudStorage == nil {
+			toUpdate.Spec.CloudStorage = &corev1.CloudStorageSpec{}
+		}
+		if toUpdate.Spec.CloudStorage.DeviceSpecs == nil {
+			defaultEKSCloudStorageDevice := "type=gp3,size=150"
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx.go:1009]: nit: Can we extract this as a constant?

File: drivers/storage/portworx/portworx.go [hunk 3]
```diff
@@ -712,14 +998,27 @@ func SetPortworxDefaults(toUpdate *corev1.StorageCluster, k8sVersion *version.Ve
...
+			deviceSpecs := []string{defaultEKSCloudStorageDevice}
+			toUpdate.Spec.CloudStorage.DeviceSpecs = &deviceSpecs
+			if toUpdate.Spec.Kvdb.Internal && toUpdate.Spec.CloudStorage.KvdbDeviceSpec == nil {
+				toUpdate.Spec.CloudStorage.KvdbDeviceSpec = &defaultEKSCloudStorageDevice
+			}
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx.go:1014]: Can we skip this? Portworx already creates a kvdb drive even if we don't specify it.

File: drivers/storage/portworx/portworx.go [hunk 4]
```diff
@@ -712,14 +998,27 @@ func SetPortworxDefaults(toUpdate *corev1.StorageCluster, k8sVersion *version.Ve
...
+		}
+		// TODO: set max storage node per zone, can we do it after or in SetDefaultsOnStorageCluster?
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx.go:1016]: What does this TODO mean?

**Review by piyush-nimbalkar — COMMENTED** [review]
Very good refactoring, @pure-jliao! 👏 
I think this is the right place for all the max storage nodes calculation.

File: drivers/storage/portworx/portworx.go (hunk 4 above)
Comment (pure-jliao) [reply drivers/storage/portworx/portworx.go:1016]: oh this was before the refactory, so already resolved.

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**